packages/common/http/src/fetch.ts

```diff
@@ -0,0 +1,230 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {inject, Injectable} from '@angular/core';
+import {BehaviorSubject, from, merge, Observable} from 'rxjs';
+import {finalize} from 'rxjs/operators';
+
+import {HttpBackend} from './backend';
+import {HttpHeaders} from './headers';
+import {HttpRequest} from './request';
+import {HttpDownloadProgressEvent, HttpErrorResponse, HttpEvent, HttpEventType, HttpHeaderResponse, HttpResponse, HttpStatusCode} from './response';
+
+const XSSI_PREFIX = /^\)\]\}',?\n/;
+
+/**
+ * Determine an appropriate URL for the response, by checking either
+ * response url or the X-Request-URL header.
+ */
+function getResponseUrl(response: Response): string|null {
+  if (response.url) {
+    return response.url;
+  }
+  // stored as lowercase in the map
+  const xRequestUrl = 'X-Request-URL'.toLocaleLowerCase();
+  return response.headers.get(xRequestUrl);
+}
+
+/**
+ * Uses `fetch` to send requests to a backend server.
+ *
+ * This `FetchBackend` requires the support of the Fetch API which is available on all evergreen
```

Comment:
nit: we can add a link here:
```suggestion
 * This `FetchBackend` requires the support of the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) which is available on all evergreen
```